packages/auto-form/src/AutoForm.tsx

```diff
@@ -0,0 +1,55 @@
+import { ConfigurationProperty } from '@syndesis/models';
+import { Formik } from 'formik';
+import * as React from 'react';
+import { FormBuilder } from './FormBuilder';
+
+export interface IFormDefinition {
+  [name: string]: ConfigurationProperty;
+}
+
+export interface IAutoFormProps {
+  definition: IFormDefinition;
+  initialValues?: object;
```

Comment:
> > I think we want both an initialValues (usually called defaultValue in react-land) and a value property here, so to allow the consumer of the component to manage the form state persistence as it works best for it. I'd drop the plural and let the type hint that.
> 
> So I understand, you're proposing a `value` property that I guess would be updated once the form definition is processed and rendered? I was actually trying to avoid that :-)
> 
> > I'm thinking about the autosave feature here, where I assume the passed in onSave callback will serialize the data on local storage, or some other medium (dunno, maybe we will have an API for this?).
> 
> Yeah, I was thinking that triggering form submission would be the way the consumer of the form library would get the value and can then do whatever at that point.

I guess it depends on when we want to autosave stuff, if while editing or at the submit moment.

For the former, we kinda have to expose the low-level API to the `AutoForm` consumer since it will probably do something also on the `handleChange` handler. 
For the latter, the way you have implemented it is fine.

I'm kinda leaning towards the first option, which is e.g. how GH implements the autosave for the comment textarea. But we can also do this in a future moment if needed.

Reply:
Ah, gotcha!  For autosave my thought was having that happen when say "next" is clicked etc.  But that doesn't capture the case like the configuration details page.  Still, for that page even there's a view mode and then an edit mode where the form is shown, so there's still going to be an "ok" button or whatever.

That makes sense though, and maybe that's the way we should go.  I think exposing a handleChange callback would be good.  Lemme see if I can get that in today otherwise I'll open an issue for it.